### DISCUSSION TITLE
Always play Auto Play animation of Animated Sprite from start

### DISCUSSION THREAD

**Discussion by hsandt** [godotengine/godot-proposals#10224] [2D]
If I'm testing sprite frames in the editor, pause the animated sprite frame in the middle of the animation (or manually set an Animation Frame in the middle) and save the scene, next time I play the animation at runtime (whether via AutoPlay or `play()` in code), **it will start mid-animation**, from the frame I saved it at.

I want to open a proposal to force Autoplay-ing Animated Sprite from the 1st frame, if autoplay is enabled. However I'm curious if some developers rely on the "play from middle of animation" behavior and whether it's worth adding a toggle option to restore the old behavior), or just always play Autoplay animation from start (which would change behavior).

This is a cheaper alternative to https://github.com/godotengine/godot-proposals/issues/6977 so I think it's worth considering.

Note: this is for AnimatedSprite2D only. Using an Animation Player will always auto-play or `play()` the animation from start (tested in Godot 4.0.2 and 4.3).

**Comment by lukaskrepel** [comment]
It would be a welcome change for my projects.

> Using an Animation Player will always auto-play or `play()` the animation from start (tested in Godot 4.0.2 and 4.3).

The fact that auto-play starts from the start in an Animation Player is enough to change the AnimatedSprite2D's behaviour to match in my opinion.

I can't imagine there's many (if any) developers who are using auto-play from mid-animation.
